src/compile/legend/parse.ts

```diff
@@ -126,6 +125,13 @@ export function parseLegendForChannel(model: UnitModel, channel: NonPositionScal
   };
 
   for (const property of LEGEND_COMPONENT_PROPERTIES) {
+    if (
+      (legendType === 'gradient' && property.startsWith('symbol')) ||
+      (legendType === 'symbol' && property.startsWith('gradient'))
+    ) {
+      continue;
```

Comment:
So we are skipping gradient properties for symbol legend and vice versa, and that's why we are seeing some symbolType and gradientLength properties being removed, right?

Reply:
Yes!